packages/blade/src/components/Accordion/AccordionButton.web.tsx

```diff
@@ -1,19 +1,28 @@
-import type { ReactElement } from 'react';
+import type { KeyboardEventHandler, ReactElement } from 'react';
 import { StyledAccordionButton } from './StyledAccordionButton';
 import type { AccordionButtonProps } from './types';
+import { useAccordion } from './AccordionContext';
 import { BaseBox } from '~components/Box/BaseBox';
 import { makeAccessible } from '~utils';
 import { Heading } from '~components/Typography';
-import { ChevronDownIcon } from '~components/Icons';
 import { useCollapsible } from '~components/Collapsible/CollapsibleContext';
+import { CollapsibleChevronIcon } from '~components/Collapsible/CollapsibleChevronIcon';
 
 const AccordionButton = ({ index, icon: Icon, children }: AccordionButtonProps): ReactElement => {
-  const { setIsExpanded } = useCollapsible();
+  const { onExpandChange, isExpanded } = useCollapsible();
+  const { showNumberPrefix, expandedIndex } = useAccordion();
 
-  const onClick = (): void => setIsExpanded((prev) => !prev);
+  const toggleCollapse = (): void => onExpandChange(!isExpanded);
+  const onClick = (): void => toggleCollapse();
+  const onKeyDown: KeyboardEventHandler<HTMLDivElement> = (event) => {
+    const SPACE_KEY = ' ';
+    if (event.key === SPACE_KEY || event.key === 'Enter') {
+      toggleCollapse();
+    }
+  };
```

Comment:
Why do we need to handle this with JavaScript? shouldn't a Button already respond to SPACE, ENTER key if we have onClick?

Reply:
I'm not using a `button` here but a `{ role: 'button' }` because we need to render `Heading` inside the accordion button. Right now since `Heading` doesn't have something like a `as="span"` it would be invalid HTML to have a `h` inside `button`.